plugins/serializers/prometheusremotewrite/prometheusremotewrite.go

```diff
@@ -52,10 +54,12 @@ func (s *Serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
 }
 
 func (s *Serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
-	var buf bytes.Buffer
+	s.buf.Reset()
```

Comment:
This function now becomes stateful and will do strange things if called concurrently. Can we please at least add a mutex to protect the buffer?

Reply:
went back to the stateless approach; it does not have big performance impacts and state is not needed.